### PR TITLE
Adding thread-safe versions for LRU cache and map

### PR DESCRIPTION
**Pull request by basharal**
- Adding thread-safe versions of LRU cache and map
- Adding Delete functionality to LRU cache

### CODE REVIEW

File: container/concurrent/lrucache.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	cache := lrucache.New(size)
+
+	return &concurrentLruCacheImp{
+		cache: cache,
+		size:  size,
+	}
+}
+
+func (p *concurrentLruCacheImp) Get(key string) (v interface{}, found bool) {
+	p.lock.RLock()
```
Comment (PatrickDropbox) [inline container/concurrent/lrucache.go]: you're r-locking twice.  once in Get and once in GetMultiple.  remove locking here

File: container/concurrent/map.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package concurrent
+
+import "sync"
+
+type Map interface {
+	Get(key interface{}) (interface{}, bool)
```
Comment (PatrickDropbox) [inline container/concurrent/map.go]: no generics =...(

File: container/concurrent/lrucache.go (hunk 1 above)
Comment (basharal) [inline container/concurrent/lrucache.go]: Good catch....I've added unit-tests too now. I guess RLock is re-entrant so, it's not as bad as a Lock

File: container/concurrent/lrucache.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+type CacheResult struct {
+	V     interface{}
+	Found bool
+}
+
+type concurrentLruCacheImp struct {
+	cache *lrucache.LRUCache
+	lock  sync.RWMutex
+	size  int
```
Comment (PatrickDropbox) [inline container/concurrent/lrucache.go]: not need to store this here.  You should expose `MaxSize() int` as part of the LRUCache interface, and expose the MaxSize from the underlying cache.

Comment (basharal) [inline container/concurrent/lrucache.go]: Done!

**Pull request merged**